agents/web-api/src/main/java/edu/jhuapl/dorset/agents/StockAgent.java

```diff
@@ -0,0 +1,228 @@
+                } catch (IOException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
+
+                }
+            }
+        }
+
+        InputStream nyseCompanies = StockAgent.class.getClassLoader().getResourceAsStream("stockagent/NYSE_Companies.csv");
+
+        try {
+            csvBeanReader = new CsvBeanReader(new BufferedReader( new InputStreamReader(nyseCompanies)),
+                    CsvPreference.STANDARD_PREFERENCE);
+            final String[] header = csvBeanReader.getHeader(true);
+            CompanyInfo companyInfo;
+            while ((companyInfo = csvBeanReader.read(CompanyInfo.class, header, processors)) != null) {
+
+                if (!this.stockSymbolMap.containsKey("apple")) {
+                    this.stockSymbolMap.put(companyInfo.getName(), companyInfo.getSymbol());
+                }
+            }
+
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        } finally {
+            if (csvBeanReader != null) {
+                try {
+                    csvBeanReader.close();
+                } catch (IOException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
+                }
+            }
+        }
+
+    }
+
+    @Override
+    public AgentResponse process(AgentRequest request) {
+
+        String requestCompanyName = request.getText().toLowerCase().replace("stocks ", "").replace("stock ", "");
+        String[] stockCompanyInfo = findStockSymbol(requestCompanyName);
+        String keywordCompanyName = stockCompanyInfo[0];
+        String keywordCompanySymbol = stockCompanyInfo[1];
+        String json = null;
+
+        if (keywordCompanySymbol == null) {
+            return new AgentResponse("I am sorry, I don't understand which company you are asking about.");
+        } else {
+            json = requestData(keywordCompanySymbol);
+        }
+
+        ArrayList<JsonElement> responseArray = new ArrayList<>();
+        ArrayList<JsonElement> responseLabelsArray = new ArrayList<>();
+
+        Gson gson = new Gson();
+
+        try {
+            JsonObject jsonObj = gson.fromJson(json.toString(),
+                    JsonObject.class);
+
+            JsonArray jsonDataArray = (JsonArray) (((JsonObject) jsonObj
+                    .get("dataset")).get("data"));
+
+            for (int i = 0; i < jsonDataArray.size(); i++) {
+                JsonArray jsonDataArrayNested = (JsonArray) (jsonDataArray
+                        .get(i));
+                responseArray.add(jsonDataArrayNested.get(4));
+                responseLabelsArray.add(jsonDataArrayNested.get(0));
+            }
+            Collections.reverse(responseArray);
+            Collections.reverse(responseLabelsArray);
+
+            List<JsonElement> returnDataArray = (responseArray.subList(
+                    responseArray.size() - 30, responseArray.size()));
```

Comment:
Changed to constant -- final int daysInAMonth = 30. 

To do: I am anticipating that a user may want to pass in the time duration (week, year, etc.), so rather than having it fixed at 30, have the user input the duration and keep the default of thirty days for cases where there are no inputs of a time duration.